src/cache.c

```diff
@@ -35,20 +35,66 @@ static pthread_mutex_t cf_lock;
  */
 static char *DATA_DIR;
 
-static char *CacheSystem_get_cache_home()
+
+/**
+ * \brief Return the fullpath to the cache or config directory
+ *
+ * \param[in] cache_or_config 1 to get cache home, 0 for config
+ */
+static char *CacheSystem_get_home(int cache_or_config)
 {
-    if (CONFIG.cache_dir) {
-        return CONFIG.cache_dir;
+    const char *xdg_home;
+    const char *xdg_default;
+    if (cache_or_config) {
+        xdg_home = "XDG_CACHE_HOME";
+        xdg_default = "/.cache";
+    } else {    
+        xdg_home = "XDG_CONFIG_HOME";
+        xdg_default = "/.config";
     }
-    char *xdg_cache_home = getenv("XDG_CACHE_HOME");
-    if (!xdg_cache_home) {
-        char *home = getenv("HOME");
-        char *xdg_cache_home_default = "/.cache";
-        xdg_cache_home = path_append(home, xdg_cache_home_default);
+
+    char *home;
+
+    if (CONFIG.cache_dir && cache_or_config) {
+        return CONFIG.cache_dir;
+    } else {
+        /*
+        *  "XDG_XYZ_HOME/HOME/.cache" > "HOME/.cache"  > "./.cache"
+        */
+        char *xdg = getenv(xdg_home);
+        if (!xdg) {
+            char *user_home = getenv("HOME");
+            if (!user_home) {
```

Comment:
Let's print a warning message when the ``$HOME`` isn't set. Let's go using ``lprintf`` at ``warning`` level.